openedx/frontend-app-learning
# feat: allow runtime configuration

`Description`

This PR is part of the work to make it possible to configure the frontend applications at runtime (you can referer to this https://github.com/openedx/frontend-wg/issues/103).

**Changes**

 - Update frontend-platform to version 2.5.0
 - Update the footer and header to avoid peer dependency errors.
  - Change the favicon in runtime.
  - Update test according to i18n v5.
  
**How to test**

- To allow runtime configuration set a new environment variables `MFE_CONFIG_API_URL` and `APP_ID` in the env file and add the api url ( To test this you can use the API from this https://github.com/openedx/edx-platform/pull/30473 or use an external tool to mock the API response).
- Due to peer dependency error with `@edx/frontend-lib-special-exams` clone the repo https://github.com/eduNEXT/frontend-lib-special-exams use the branch `dcoa/update-fronted-platform` and use webpack local module configuration ([reference](https://github.com/openedx/frontend-build#local-module-configuration-for-webpack)) to test it.

```javascript
module.exports = {
    localModules: [
        { moduleName: '@edx/frontend-lib-special-exams', dir: '../frontend-lib-special-exams', dist: 'dist' },
    ],
};

```
- The API should respond with a JSON with the config values, something like:
```json
{
  "SITE_NAME": "Test Site",
  "LOGO_URL": "https://testimage.com/logo.svg",
  "LOGO_TRADEMARK_URL": "https://testimage.com/logo.svg",
  "LOGO_WHITE_URL": "https://testimage.com/logo.svg",
  "FAVICON_URL": "https://testimage.com/favicon.ico",
}
```
- The initialize process should work normally.
_Note: You can combine buildtime and runtime configuration_
